src/common/snippets/include/snippets/lowered/pass/brgemm_debug_params.hpp

```diff
@@ -0,0 +1,155 @@
+            auto params = collect_params(brgemm_expr, linear_ir);
+            const auto& perf_count_begin = std::make_shared<snippets::op::PerfCountBegin>();
+            perf_count_begin->set_friendly_name(std::string("PerfCount_Begin_") + std::to_string(seq_number) +
+                                                "_DebugParams");
+            const auto empty_inputs = std::vector<PortConnectorPtr>{};
+            linear_ir.insert_node(perf_count_begin, empty_inputs, expr_it->get()->get_loop_ids(), false, expr_it);
+
+            const auto& perf_count_end = std::make_shared<snippets::op::PerfCountEnd>(perf_count_begin->output(0));
+            perf_count_end->set_friendly_name(std::string("PerfCount_End_") + std::to_string(seq_number) +
+                                              "_DebugParams");
+            // Attach brgemm parameters to PerfCountEnd node
+            auto csv_path = linear_ir.get_config().debug_config.dumpParams.csv_path;
+            perf_count_end->get_rt_info()["brgemm_params"] = params;
+            perf_count_end->get_rt_info()["brgemm_params_csv_path"] = csv_path;
+            linear_ir.insert_node(perf_count_end, empty_inputs, expr_it->get()->get_loop_ids(), false, next(expr_it));
+            seq_number++;
+            modified = true;
+        }
+        return modified;
+    }
+
+private:
+    std::string collect_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr,
+                               const snippets::lowered::LinearIR& linear_ir) {
+        auto debug_config = linear_ir.get_config().debug_config;
+        const auto brgemm = ov::as_type_ptr<BRGEMM_TYPE>(brgemm_expr->get_node());
+        OPENVINO_ASSERT(brgemm, "Brgemm is nullptr!");
+        std::stringstream ss;
+        ss << brgemm_expr->get_node()->get_friendly_name() << ',';
+        ss << m_subgraph_name << ',';
+        for (size_t i = 0; i < brgemm->get_input_size(); ++i) {
+            ss << brgemm->get_input_element_type(i);
+            if (i != brgemm->get_input_size() - 1) {
+                ss << ';';
+            }
+        }
+        ss << ',';
+        for (size_t i = 0; i < brgemm->get_output_size(); ++i) {
+            ss << brgemm->get_output_element_type(i);
+            if (i != brgemm->get_output_size() - 1) {
+                ss << ';';
+            }
+        }
+        ss << ',';
+        auto append_shape_info = [&](const auto& ports, auto get_port_descriptor) {
+            size_t i = 0;
+            for (const auto& port : ports) {
+                if (port.get_partial_shape().is_dynamic()) {
+                    const auto& partial_shape = port.get_partial_shape();
+                    for (const auto& dim : partial_shape) {
+                        ss << dim << ' ';
+                    }
+                } else {
+                    const auto& shape =
+                        ov::snippets::utils::get_planar_vdims(port.get_shape(), get_port_descriptor(i)->get_layout());
```

Comment:
Please use get_port_descriptor(i)->get_shape() instead -- this will work for both static and dynamic shapes
We don't need brgemm inputs/outputs at all: all the necessary info is stored in port descriptors

Reply:
Done